### PR TITLE
[ADS-31014] Adds swagger_op_id_formatter config to apipie

### PR DESCRIPTION
**Pull request by jimmypitts92**
Adding a formatter class for the generation of swagger_op_ids.  This allows us control over the formatting.

The `path` is a representation of an endpoint, Ie. `put_ad_group_keywords_id`
The `method` is used as a title in request bodies.

We will only be overriding the `format_op_id_for_method` method.  By default, that takes the controller path (including versioning info).  Ie. `Api__V1__AdGroupProductsController_Input`.

This will allow us to format these strings to remove any of the versioning info.  This will allow our OpenAPI spec (and therefor frontend client) to remain unchanged throughout version upgrades.

This follows the convention of the existing `Apipie::RoutesFormatter` class.

Initial plan for what our override class in ads-backend will look like:

```
# frozen_string_literal: true

module Api::Master
  module Concerns
    class ApiSwaggerOpIdFormatter < Apipie::SwaggerOpIdFormatter
      def format_op_id_for_method(op_id)
        op_id = super
        # Pulls `__Master` or `__Vx` from `Api__V1__AdGroupProductsController_Input`
        regex = /^Api(__[vV]\d)?(__[Mm]aster)?/
        op_id = op_id.gsub(regex, 'Api')
        op_id
      end
    end
  end
end
```

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -991,6 +991,11 @@ routes_formatter
   format usable in the documentation when using the `api!` keyword. By
   default, the ``Apipie::RoutesFormatter`` is used.
 
+swagger_op_id_formatter
```
Comment (AzRu) [inline README.rst:994]: looks fine but got me wondering 'what is an op id'

File: lib/apipie/swagger_generator.rb [hunk 2]
```diff
@@ -269,7 +271,9 @@ def remove_colons(str)
     end
 
     def swagger_op_id_for_method(method)
-      remove_colons method.resource.controller.name + '::' + method.method
+      op_id = remove_colons method.resource.controller.name + '::' + method.method
```
Comment (emmettu) [inline lib/apipie/swagger_generator.rb:274]: nit: Seems like we'd have more flexibility if this part was also done in the OpIdFormatter and it just generated the whole op_id from the method/path.

**Review by emmettu — APPROVED** [review]
LGTM, one comment

File: lib/apipie/swagger_generator.rb (hunk 2 above)
Comment (jimmypitts92) [reply lib/apipie/swagger_generator.rb:274]: I originally had this in there, and then pulled it out as I think it's a more flexible approach.  1 - it follows the same convention as the RoutesFormatter, where the route is already formatted a standard way, and then you can add a handler to provide additional formatting.  2 - this is a little more extensible.  This approach reduces the amount of changes to the core apipie functionality.  If we want to upgrade apipie versions, having this logic not exist in the formatter will make that easier.

Comment (jimmypitts92) [reply lib/apipie/swagger_generator.rb:274]: Scratch point 1, but point 2 still stands

File: README.rst (hunk 1 above)
Comment (jimmypitts92) [reply README.rst:994]: It's just what apipie uses to identify certain operations in the openapie spec.  Specifically, these are `path` or `method` identifiers (paths are routes), and methods are input param groupings.

**Pull request merged**